server/src/main/java/org/apache/druid/server/http/MetadataResource.java

```diff
@@ -165,6 +173,104 @@ public Response getAllUsedSegments(
     return builder.entity(authorizedSegments).build();
   }
 
+  @GET
+  @Path("/changedSegments")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getChangedSegments(
```

Comment:
Could you document what each parameter is supposed to be? I think it's not clear from the name and it's hard to tell what they should be without going through ChangeRequestHistory.